### PR TITLE
#28 flatten nested arrays

### PR DESCRIPTION
**Pull request by lukaszkobylecki**

### CODE REVIEW

**Comment by lukaszkobylecki** [comment]
Resolves #28

File: src/processing/preprocess.js [hunk 1]
```diff
@@ -18,4 +18,49 @@ const preprocess = (json) => {
...
+
+  return processed;
+};
+
+const preprocess = (json) => {
+  let preprocessedJson = json;
+
+  Logger.log(new LogMessage('Preprocessing json object.'));
+
+  preprocessedJson = handleSimpleTypes(preprocessedJson);
```
Comment (tomaszgil) [inline src/processing/preprocess.js]: Instead of going through json 3 times we should do it all in one run.

File: src/processing/preprocess.js [hunk 2]
```diff
@@ -18,4 +18,49 @@ const preprocess = (json) => {
   return processed;
 };
 
+const flattenArrays = (json) => {
+  const processed = json;
+
+  Object.keys(processed).forEach((key) => {
+    if (Array.isArray(processed[key])) {
+      processed[key] = [].concat(...processed[key]);
```
Comment (tomaszgil) [inline src/processing/preprocess.js]: `processed[key] = [...processed[key]];`

Comment (lukaszkobylecki) [reply src/processing/preprocess.js]: It is not the same

**Pull request merged**